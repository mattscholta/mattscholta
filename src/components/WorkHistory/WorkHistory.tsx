import * as React from 'react';
import classnames from 'classnames';

import {
  data as experience,
  Experience as ExperienceInstance
} from '@src/data/experience';

// import { analytics } from '@client/utils/analytics';
import styles from './styles.module.css';

export interface WorkHistoryProps {
  className?: string;
}

/**
 * @name WorkHistory
 * @description Stateless Component / Pure Function
 */
const WorkHistory: React.FC<WorkHistoryProps> = (props) => {
  const { className } = props;

  // Styles
  const css = classnames(styles.component, className);

  // Handlers
  const onClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    const { currentTarget } = event;
    const { gaLabel } = currentTarget.dataset;

    // analytics.event('work-history', 'click', gaLabel);
  };

  // Markup
  const renderWorkplace = (node: ExperienceInstance) => {
    const { date, description, featured, link, name, role } = node;
    const cssWorkplace = classnames({ featured });

    return (
      <li className={cssWorkplace} key={link}>
        <h3>
          <a
            className="spot-primary"
            data-ga-label={name.toLowerCase()}
            href={link}
            onClick={onClick}
            rel="noopener noreferrer"
            target="_blank"
          >
            {name}
          </a>{' '}
          | {role}
        </h3>
        <em>{date}</em>
        {/* <div>{description}</div> */}
      </li>
    );
  };

  return (
    <div className={css}>
      <h2 className="ui-heading medium">Work History / Experience</h2>
      <ul className={styles.workplaces}>{experience.map(renderWorkplace)}</ul>
      <p className={styles.more}>
        * Visit{' '}
        <a
          className="spot-primary"
          href="https://www.linkedin.com/in/matthewscholta"
          rel="noopener noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>{' '}
        for more info.
      </p>
    </div>
  );
};

export { WorkHistory };
