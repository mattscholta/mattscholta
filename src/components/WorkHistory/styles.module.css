.component,
.component h2 {
  margin-bottom: var(--grid-4x);
}

.component h5 {
  margin-bottom: var(--grid-1xs);
}

.component div {
  font-size: 0.9em;
  margin-top: var(--grid-2xs);
}

.component em {
  font-size: 0.8em;
  padding-left: var(--grid-2x);
}

.component li {
  border: 1px solid var(--spot-grey-1);
  margin-bottom: var(--grid-2x);
  padding: var(--grid-2x);
}

.component li:global(.featured) {
  border-left: var(--grid-1x) solid var(--spot-grey-1);
}

.component li:global(.featured) em {
  color: var(--spot-primary);
  font-weight: bold;
}

.more {
  font-size: 0.8em;
  font-style: italic;
  text-align: right;
}

.workplaces h3 {
  font-size: 1em;
  font-weight: 100;
  text-transform: none;
}
