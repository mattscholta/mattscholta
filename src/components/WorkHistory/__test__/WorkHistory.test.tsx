import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { WorkHistory, WorkHistoryProps } from '../WorkHistory';

describe('WorkHistory', () => {
  let props: WorkHistoryProps;
  let wrapper: RenderResult;

  const createWrapper = () => <WorkHistory {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
