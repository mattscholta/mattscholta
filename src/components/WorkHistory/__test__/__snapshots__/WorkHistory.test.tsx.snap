// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkHistory matches snapshot 1`] = `
NodeList [
  <div
    class="component"
  >
    <h2
      class="ui-heading medium"
    >
      Work History / Experience
    </h2>
    <ul
      class="workplaces"
    >
      <li
        class="featured"
      >
        <h3>
          <a
            class="spot-primary"
            data-ga-label="haldi"
            href="https://haldi.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Haldi
          </a>
           
          | 
          Principal Engineer
        </h3>
        <em>
          April 2021 – Present
        </em>
      </li>
      <li
        class=""
      >
        <h3>
          <a
            class="spot-primary"
            data-ga-label="thredup"
            href="https://www.thredup.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            ThredUP
          </a>
           
          | 
          Staff Engineer
        </h3>
        <em>
          Feb 2016 – April 2021
        </em>
      </li>
      <li
        class=""
      >
        <h3>
          <a
            class="spot-primary"
            data-ga-label="beyond"
            href="https://www.bynd.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Beyond
          </a>
           
          | 
          Technical Team Lead
        </h3>
        <em>
          Nov 2011 – Apr 2015
        </em>
      </li>
    </ul>
    <p
      class="more"
    >
      * Visit
       
      <a
        class="spot-primary"
        href="https://www.linkedin.com/in/matthewscholta"
        rel="noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
       
      for more info.
    </p>
  </div>,
]
`;
